### PR TITLE
schemast: Support Mixins for UpsertSchema

### PR DESCRIPTION
**Pull request by C-Deck**
* Add mixin kind
* Add AppendMixin to Context
* Add Mixin func that provides a ast.Expr

### CODE REVIEW

File: schemast/mixin.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package schemast
```
Comment (rotemtam) [inline schemast/mixin.go:1]: add license headers in all new files

**Comment by rotemtam** [comment]
@C-Deck 👋

Did you see my last review comments ? I think we're close to done here 😊

**Comment by C-Deck** [comment]
> @C-Deck 👋
> 
> Did you see my last review comments ? I think we're close to done here 😊

I have been unable to come back to this. I will update the PR with changes per the review.

File: schemast/internal/mixintest/nanoid.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package mixintest
```
[the remark this answers is not in the document]
Comment (C-Deck) [reply schemast/internal/mixintest/nanoid.go:1]: Meant to do uuid.go. Updating.

File: schemast/mixin.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+func (c *Context) AppendMixin(typeName string, mixin ent.Mixin) error {
+	newMixin, importPkg, err := Mixin(mixin)
+	if err != nil {
+		return err
+	}
+	c.appendImport(typeName, importPkg)
+	return c.appendReturnItem(kindMixin, typeName, newMixin)
+}
+
+func Mixin(mixin ent.Mixin) (ast.Expr, string, error) {
```
[the remark this answers is not in the document]
Comment (C-Deck) [reply schemast/mixin.go:21]: For other things in the `UpsertSchema`, these methods are exported. I exported it to continue with that pattern.

```
func Index(desc *index.Descriptor) (*ast.CallExpr, error) {
...
}
```

```
func Field(desc *field.Descriptor) (*ast.CallExpr, error) {
...
}
```

In response to the second part, can you give an example of how to infer it via the returned ast.Expr? I do not know how to do it unfortunately.

**Pull request closed without merging**